src/auth/downloaderWrapper.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <QObject>
+#include <QMap>
+
+QT_BEGIN_NAMESPACE
+class Downloader;
+class QUrl;
+QT_END_NAMESPACE
+
+class DownloaderWrapper : public QObject {
+Q_OBJECT
+
+private:
+    Downloader*   mpDownloader;
+
+    QString mUserName;
+    QString mPassword;
+    QString mToken;
+    QString mUrl;
+    QByteArray mData;
+
+public:
+    DownloaderWrapper(QObject* pwgt = nullptr);
+
+    void get( const QString &userName,        const QString &url,

```

Comment:
`void get (const QString &userName, const QString &url, const QString &tokenOrPassword = "");`
Isn't it  better?

Reply:
remade